src/unit/test_fifo.c

```diff
@@ -0,0 +1,261 @@
+#include "../fifo.c"
+#include "test_help.h"
+#include <sys/wait.h>
+#include <unistd.h>
+
+static inline void *INT_TO_PTR(intptr_t i) {
+    return (void *)i;
+}
+
+static inline intptr_t PTR_TO_INT(void *p) {
+    return (intptr_t)p;
+}
+
+/* Helper functions */
+static void push(Fifo *q, intptr_t value) {
+    int len = fifoLength(q);
+    fifoPush(q, INT_TO_PTR(value));
+    TEST_EXPECT(fifoLength(q) == len + 1);
+}
+
+static intptr_t popTest(Fifo *q, intptr_t expected) {
+    intptr_t peekValue = PTR_TO_INT(fifoPeek(q));
+    TEST_EXPECT(peekValue == expected);
+    int len = fifoLength(q);
+    intptr_t value = PTR_TO_INT(fifoPop(q));
+    TEST_EXPECT(fifoLength(q) == len - 1);
+    TEST_EXPECT(value == expected);
+    return value;
+}
+
+/* Test: emptyPop - verify that popping from empty queue triggers assertion */
+int test_fifoEmptyPop(int argc, char *argv[], int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    pid_t pid = fork();
+    if (pid == 0) {
+        /* Child process - this should crash with assertion */
+        Fifo *q = fifoCreate();
+        TEST_EXPECT(fifoLength(q) == 0);
+        fifoPop(q); /* This should assert and crash */
+        /* Should never reach here */
+        fifoDelete(q);
+        exit(0);
+    } else if (pid > 0) {
+        /* Parent process - wait for child to crash */
+        int status;
+        waitpid(pid, &status, 0);
+        /* Verify child exited abnormally (assertion failure) */
+        TEST_EXPECT(WIFSIGNALED(status) || (WIFEXITED(status) && WEXITSTATUS(status) != 0));
+    } else {
+        /* Fork failed */
+        TEST_EXPECT(0);
+    }
+    return 0;
+}
+
+/* Test: emptyPeek - verify that peeking at empty queue triggers assertion */
+int test_fifoEmptyPeek(int argc, char *argv[], int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    pid_t pid = fork();
+    if (pid == 0) {
+        /* Child process - this should crash with assertion */
+        Fifo *q = fifoCreate();
+        TEST_EXPECT(fifoLength(q) == 0);
+        fifoPeek(q); /* This should assert and crash */
+        /* Should never reach here */
+        fifoDelete(q);
+        exit(0);
+    } else if (pid > 0) {
+        /* Parent process - wait for child to crash */
+        int status;
+        waitpid(pid, &status, 0);
+        /* Verify child exited abnormally (assertion failure) */
+        TEST_EXPECT(WIFSIGNALED(status) || (WIFEXITED(status) && WEXITSTATUS(status) != 0));
```

Comment:
Since we do this twice, maybe create a function `detectChildCrash`?

Edit:  Even better, how about a macro like:
```c
DETECT_CRASH {
}
```
This could expand into:
```c
    pid_t pid = fork();
    if (pid < 0) {
        /* Fork failed */
        TEST_EXPECT(0);
    } else if (pid > 0) {
        /* Parent process - wait for child to crash */
        int status;
        waitpid(pid, &status, 0);
        /* Verify child exited abnormally (assertion failure) */
        TEST_EXPECT(WIFSIGNALED(status) || (WIFEXITED(status) && WEXITSTATUS(status) != 0));
    } else
```